teps/0033-tekton-feature-gates.md

```diff
@@ -145,6 +160,393 @@ alpha features. This is not a requirement for Tekton, because even in case of
+alpha field, the webhook will not allow the CRD to be stored and will return an actionable error.
+(Same for beta.)
+
+Pros:
+* One flag (for each of the 2 levels of stability) makes it easy to opt into alpha features
+  as a whole (assumes a user who wants any of these features wants "cutting edge" features in general)
+* Provides a simpler testing matrix than one flag per field
+  (see [one feature flag per new alpha field](#one-feature-flag-per-new-alpha-field)
+  alternative for details)
+
+Cons:
+* Users who want to use new functionality and do not have admin access to their clusters will
+  not be able to (unless the folks with admin access let them)
+* If someone adds a new field and it is not added as optional, removing it will be a backwards
+  incompatible change even for users who do not opt into alpha features
+* Removing a field that is considered "alpha" or "beta" may still be perceived as a
+  backwards incompatible change
+    * This is allowed by our policy because the field is considered alpha (or beta)
+* If alpha fields are always enabled on a cluster, a user might not realize they have started to rely
+  on an alpha feature
+    * In the future we may try to communicate this to the user somehow, e.g. through something in the
+      run status (probably through logs at the very least)
+* Tools that integrate with Tekton may have a hard time telling the difference between alpha
+  fields and beta/v1, e.g. a vscode integration with auto completion
+    * Hopefully we can mitigate this with documentation; it will be up to the integrating tools
+      to decide how to handle this
+* These two fields would toggle alpha and beta fields across all CRDs; another option would be
+  to have a flag per version per CRD
+
+#### Docs
+
+We will maintain documentation showing the status of each field:
+
+* We will create a table of fields that are considered alpha or beta (which will link to docs
+  on each feature)
+* In the docs for each CRD, we will have a section for alpha fields and beta fields
+
+### Changes in behavior
+
+Changes in behavior are features that are not controlled by CRD fields, for example if we
+wanted to change finally tasks such that they always run when a PipelineRun is cancelled.
+
+When we need to make behavior changes like this, each will get its own
+[feature flag](https://github.com/tektoncd/pipeline/blob/master/docs/install.md#customizing-the-pipelines-controller-behavior),
+like existing feature flags such as `disable-affinity-assistant` and `disable-home-env-overwrite`.
+
+_We should avoid adding features like this if possible since each new feature
+complicates the combination of settings we'll need to test in order to feel
+confident in our automated testing, and because promoting these changes from
+alpha will be a backwards incompatible change._ 
+
+Pros:
+* It will be clear to users whether they have opted into this functionality or not
```

Comment:
I think the meaning of "users" here is cluster admins, is that right?  I'm not sure it'll be clear to non-admin users?